### ISSUE TITLE
[Doc]: How to make Multi-Node Inference

### ISSUE THREAD

**Issue by pygongnlp** [vllm-project/vllm#10886]
### 📚 The doc issue

I need to use 72B model for inference about 10W cases, and have learn about Distributed Inference and Serving.
I have 4 machines, each with 2 A10 GPUs. I think vLLM can split this model across 8 GPUs using tensor parallelism and pipeline parallelism. (Ray can start on these machines)

However, I haven't found how to do multi-node inference. The documentation seems to only cover single-node inference & serving, as well as multi-node serving, which might not be what I need.

### Suggest a potential alternative/fix

If can rewrite to add how to make multi-node inference

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by jikunshang** [comment]
I think you can follow this part https://docs.vllm.ai/en/latest/serving/distributed_serving.html#multi-node-inference-and-serving

**Comment by pygongnlp** [comment]
> I think you can follow this part https://docs.vllm.ai/en/latest/serving/distributed_serving.html#multi-node-inference-and-serving

thanks a lot
Is there a way to perform multi-node offline inference rather than make a serving?

**Comment by jikunshang** [comment]
> > I think you can follow this part https://docs.vllm.ai/en/latest/serving/distributed_serving.html#multi-node-inference-and-serving
> 
> thanks a lot Is there a way to perform multi-node offline inference rather than make a serving?

I am afraid the answer is no. Is there any specific scenario for offline inference?

**Comment by pygongnlp** [comment]
> > > I think you can follow this part https://docs.vllm.ai/en/latest/serving/distributed_serving.html#multi-node-inference-and-serving
> > 
> > 
> > thanks a lot Is there a way to perform multi-node offline inference rather than make a serving?
> 
> I am afraid the answer is no. Is there any specific scenario for offline inference?



> > > I think you can follow this part https://docs.vllm.ai/en/latest/serving/distributed_serving.html#multi-node-inference-and-serving
> > 
> > 
> > thanks a lot Is there a way to perform multi-node offline inference rather than make a serving?
> 
> I am afraid the answer is no. Is there any specific scenario for offline inference?

Thanks for your reply.

The situation on our side might be like this: we want to use a 72B size model for inference on massive amounts of data, possibly tens of millions of samples.

We are performing this operation on 4 machines, with 2 A10 GPUs per machine, so we are using --tensor-parallel-size 2 and --pipeline-parallel-size 4.

We then found that LLMEngine does not support pipeline-parallel-size, but AsyncLLMEngine does.

As for why we want to use offline inference, perhaps it's because it's simpler this way? I may not be very familiar with efficient inference for LLMs, so I chose this approach, hoping to speed up inference by increasing the batch size during the inference phase.

We also tried using vllm serve, but we encountered the following timeout issue (which was previously resolved by upgrading the requests and ray versions, but it keeps reoccurring). 

<details><summary>Details</summary>
<p>

[rank0]: Traceback (most recent call last):
[rank0]:   File "/root/miniconda3/envs/pygong/bin/vllm", line 8, in <module>
[rank0]:     sys.exit(main())
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/scripts.py", line 195, in main
[rank0]:     args.dispatch_function(args)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/scripts.py", line 41, in serve
[rank0]:     uvloop.run(run_server(args))
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/uvloop/__init__.py", line 82, in run
[rank0]:     return loop.run_until_complete(wrapper())
[rank0]:   File "uvloop/loop.pyx", line 1518, in uvloop.loop.Loop.run_until_complete
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/uvloop/__init__.py", line 61, in wrapper
[rank0]:     return await main
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/entrypoints/openai/api_server.py", line 609, in run_server
[rank0]:     async with build_async_engine_client(args) as engine_client:
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/contextlib.py", line 181, in __aenter__
[rank0]:     return await self.gen.__anext__()
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/entrypoints/openai/api_server.py", line 113, in build_async_engine_client
[rank0]:     async with build_async_engine_client_from_engine_args(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/contextlib.py", line 181, in __aenter__
[rank0]:     return await self.gen.__anext__()
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/entrypoints/openai/api_server.py", line 146, in build_async_engine_client_from_engine_args
[rank0]:     engine_client = build_engine()
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/engine/async_llm_engine.py", line 691, in from_engine_args
[rank0]:     engine = cls(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/engine/async_llm_engine.py", line 578, in __init__
[rank0]:     self.engine = self._engine_class(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/engine/async_llm_engine.py", line 264, in __init__
[rank0]:     super().__init__(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/engine/llm_engine.py", line 347, in __init__
[rank0]:     self.model_executor = executor_class(vllm_config=vllm_config, )
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/executor/ray_gpu_executor.py", line 512, in __init__
[rank0]:     super().__init__(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/executor/distributed_gpu_executor.py", line 26, in __init__
[rank0]:     super().__init__(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/executor/executor_base.py", line 36, in __init__
[rank0]:     self._init_executor()
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/executor/ray_gpu_executor.py", line 65, in _init_executor
[rank0]:     self._init_workers_ray(placement_group)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/executor/ray_gpu_executor.py", line 280, in _init_workers_ray
[rank0]:     self._run_workers("load_model",
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/executor/ray_gpu_executor.py", line 424, in _run_workers
[rank0]:     ray_worker_outputs = ray.get(ray_worker_outputs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/ray/_private/auto_init_hook.py", line 21, in auto_init_wrapper
[rank0]:     return fn(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/ray/_private/client_mode_hook.py", line 103, in wrapper
[rank0]:     return func(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/ray/_private/worker.py", line 2755, in get
[rank0]:     values, debugger_breakpoint = worker.get_objects(object_refs, timeout=timeout)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/ray/_private/worker.py", line 906, in get_objects
[rank0]:     raise value.as_instanceof_cause()
[rank0]: ray.exceptions.RayTaskError(ConnectTimeout): ray::RayWorkerWrapper.execute_method() (pid=11613, ip=11.46.89.2, actor_id=1ee3417e588d766de2c0effe05000000, repr=<vllm.executor.ray_utils.RayWorkerWrapper object at 0x7f6e0bd11340>)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/connection.py", line 85, in create_connection
[rank0]:     raise err
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection
[rank0]:     sock.connect(sa)
[rank0]: socket.timeout: timed out

[rank0]: The above exception was the direct cause of the following exception:

[rank0]: ray::RayWorkerWrapper.execute_method() (pid=11613, ip=11.46.89.2, actor_id=1ee3417e588d766de2c0effe05000000, repr=<vllm.executor.ray_utils.RayWorkerWrapper object at 0x7f6e0bd11340>)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
[rank0]:     response = self._make_request(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 490, in _make_request
[rank0]:     raise new_e
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 466, in _make_request
[rank0]:     self._validate_conn(conn)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
[rank0]:     conn.connect()
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connection.py", line 693, in connect
[rank0]:     self.sock = sock = self._new_conn()
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connection.py", line 208, in _new_conn
[rank0]:     raise ConnectTimeoutError(
[rank0]: urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x7f6db0032d30>, 'Connection to huggingface.co timed out. (connect timeout=10)')

[rank0]: The above exception was the direct cause of the following exception:

[rank0]: ray::RayWorkerWrapper.execute_method() (pid=11613, ip=11.46.89.2, actor_id=1ee3417e588d766de2c0effe05000000, repr=<vllm.executor.ray_utils.RayWorkerWrapper object at 0x7f6e0bd11340>)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/adapters.py", line 667, in send
[rank0]:     resp = conn.urlopen(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 843, in urlopen
[rank0]:     retries = retries.increment(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/retry.py", line 519, in increment
[rank0]:     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
[rank0]: urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models/Qwen/Qwen2.5-0.5B-Instruct (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f6db0032d30>, 'Connection to huggingface.co timed out. (connect timeout=10)'))

[rank0]: During handling of the above exception, another exception occurred:

[rank0]: ray::RayWorkerWrapper.execute_method() (pid=11613, ip=11.46.89.2, actor_id=1ee3417e588d766de2c0effe05000000, repr=<vllm.executor.ray_utils.RayWorkerWrapper object at 0x7f6e0bd11340>)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/worker_base.py", line 481, in execute_method
[rank0]:     raise e
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/worker_base.py", line 472, in execute_method
[rank0]:     return executor(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/worker.py", line 152, in load_model
[rank0]:     self.model_runner.load_model()
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/model_runner.py", line 1074, in load_model
[rank0]:     self.model = get_model(vllm_config=self.vllm_config)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/__init__.py", line 12, in get_model
[rank0]:     return loader.load_model(vllm_config=vllm_config)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 334, in load_model
[rank0]:     model.load_weights(self._get_all_weights(model_config, model))
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/qwen2.py", line 503, in load_weights
[rank0]:     loader.load_weights(weights)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 229, in load_weights
[rank0]:     autoloaded_weights = list(self._load_module("", self.module, weights))
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 181, in _load_module
[rank0]:     for child_prefix, child_weights in self._groupby_prefix(weights):
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 103, in _groupby_prefix
[rank0]:     for prefix, group in itertools.groupby(weights_by_parts,
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 100, in <genexpr>
[rank0]:     weights_by_parts = ((weight_name.split(".", 1), weight_data)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 313, in _get_all_weights
[rank0]:     yield from self._get_weights_iterator(primary_weights)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 272, in _get_weights_iterator
[rank0]:     hf_folder, hf_weights_files, use_safetensors = self._prepare_weights(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 228, in _prepare_weights
[rank0]:     hf_folder = download_weights_from_hf(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/weight_utils.py", line 234, in download_weights_from_hf
[rank0]:     file_list = fs.ls(model_name_or_path, detail=False, revision=revision)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 280, in ls
[rank0]:     resolved_path = self.resolve_path(path, revision=revision)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 186, in resolve_path
[rank0]:     repo_and_revision_exist, err = self._repo_and_revision_exist(repo_type, repo_id, revision)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 121, in _repo_and_revision_exist
[rank0]:     self._api.repo_info(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
[rank0]:     return fn(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 2756, in repo_info
[rank0]:     return method(
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
[rank0]:     return fn(*args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 2540, in model_info
[rank0]:     r = get_session().get(path, headers=headers, timeout=timeout, params=params)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 602, in get
[rank0]:     return self.request("GET", url, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
[rank0]:     resp = self.send(prep, **send_kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
[rank0]:     r = adapter.send(request, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/utils/_http.py", line 93, in send
[rank0]:     return super().send(request, *args, **kwargs)
[rank0]:   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/adapters.py", line 688, in send
[rank0]:     raise ConnectTimeout(e, request=request)
[rank0]: requests.exceptions.ConnectTimeout: (MaxRetryError("HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models/Qwen/Qwen2.5-0.5B-Instruct (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f6db0032d30>, 'Connection to huggingface.co timed out. (connect timeout=10)'))"), '(Request ID: c0e07c68-cfb3-4f08-88c3-9f26f07e3409)')
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] Error executing method load_model. This might cause deadlock in distributed execution.
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] Traceback (most recent call last):
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connection.py", line 199, in _new_conn
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     sock = connection.create_connection(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/connection.py", line 85, in create_connection
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     raise err
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     sock.connect(sa)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] socket.timeout: timed out
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] 
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] The above exception was the direct cause of the following exception:
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] 
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] Traceback (most recent call last):
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 789, in urlopen
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     response = self._make_request(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 490, in _make_request
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     raise new_e
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 466, in _make_request
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     self._validate_conn(conn)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     conn.connect()
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connection.py", line 693, in connect
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     self.sock = sock = self._new_conn()
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connection.py", line 208, in _new_conn
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     raise ConnectTimeoutError(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x7f6db0032d30>, 'Connection to huggingface.co timed out. (connect timeout=10)')
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] 
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] The above exception was the direct cause of the following exception:
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] 
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] Traceback (most recent call last):
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/adapters.py", line 667, in send
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     resp = conn.urlopen(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 843, in urlopen
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     retries = retries.increment(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/retry.py", line 519, in increment
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models/Qwen/Qwen2.5-0.5B-Instruct (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f6db0032d30>, 'Connection to huggingface.co timed out. (connect timeout=10)'))
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] 
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] During handling of the above exception, another exception occurred:
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] 
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] Traceback (most recent call last):
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/worker_base.py", line 472, in execute_method
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     return executor(*args, **kwargs)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/worker.py", line 152, in load_model
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     self.model_runner.load_model()
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/model_runner.py", line 1074, in load_model
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     self.model = get_model(vllm_config=self.vllm_config)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/__init__.py", line 12, in get_model
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     return loader.load_model(vllm_config=vllm_config)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 334, in load_model
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     model.load_weights(self._get_all_weights(model_config, model))
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/qwen2.py", line 503, in load_weights
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     loader.load_weights(weights)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 229, in load_weights
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     autoloaded_weights = list(self._load_module("", self.module, weights))
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 181, in _load_module
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     for child_prefix, child_weights in self._groupby_prefix(weights):
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 103, in _groupby_prefix
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     for prefix, group in itertools.groupby(weights_by_parts,
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 100, in <genexpr>
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     weights_by_parts = ((weight_name.split(".", 1), weight_data)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 313, in _get_all_weights
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     yield from self._get_weights_iterator(primary_weights)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 272, in _get_weights_iterator
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     hf_folder, hf_weights_files, use_safetensors = self._prepare_weights(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 228, in _prepare_weights
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     hf_folder = download_weights_from_hf(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/weight_utils.py", line 234, in download_weights_from_hf
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     file_list = fs.ls(model_name_or_path, detail=False, revision=revision)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 280, in ls
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     resolved_path = self.resolve_path(path, revision=revision)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 186, in resolve_path
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     repo_and_revision_exist, err = self._repo_and_revision_exist(repo_type, repo_id, revision)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 121, in _repo_and_revision_exist
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     self._api.repo_info(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     return fn(*args, **kwargs)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 2756, in repo_info
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     return method(
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     return fn(*args, **kwargs)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 2540, in model_info
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     r = get_session().get(path, headers=headers, timeout=timeout, params=params)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 602, in get
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     return self.request("GET", url, **kwargs)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 589, in request
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     resp = self.send(prep, **send_kwargs)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 703, in send
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     r = adapter.send(request, **kwargs)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/utils/_http.py", line 93, in send
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     return super().send(request, *args, **kwargs)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/adapters.py", line 688, in send
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480]     raise ConnectTimeout(e, request=request)
(RayWorkerWrapper pid=11613, ip=11.46.89.2) ERROR 12-07 20:30:45 worker_base.py:480] requests.exceptions.ConnectTimeout: (MaxRetryError("HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models/Qwen/Qwen2.5-0.5B-Instruct (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f6db0032d30>, 'Connection to huggingface.co timed out. (connect timeout=10)'))"), '(Request ID: c0e07c68-cfb3-4f08-88c3-9f26f07e3409)')
(RayWorkerWrapper pid=15117, ip=11.46.94.229) INFO 12-07 20:30:34 selector.py:135] Using Flash Attention backend. [repeated 6x across cluster] (Ray deduplicates logs by default. Set RAY_DEDUP_LOGS=0 to disable log deduplication, or see https://docs.ray.io/en/master/ray-observability/user-guides/configure-logging.html#log-deduplication for more options.)
(RayWorkerWrapper pid=65163) INFO 12-07 20:30:35 utils.py:961] Found nccl from library libnccl.so.2 [repeated 13x across cluster]
(RayWorkerWrapper pid=65163) INFO 12-07 20:30:35 pynccl.py:69] vLLM is using nccl==2.21.5 [repeated 13x across cluster]
(RayWorkerWrapper pid=65163) INFO 12-07 20:30:35 custom_all_reduce_utils.py:242] reading GPU P2P access cache from /root/.cache/vllm/gpu_p2p_access_cache_for_0,1.json [repeated 6x across cluster]
(RayWorkerWrapper pid=11523, ip=11.46.89.2) INFO 12-07 20:30:35 shm_broadcast.py:236] vLLM message queue communication handle: Handle(connect_ip='127.0.0.1', local_reader_ranks=[1], buffer=<vllm.distributed.device_communicators.shm_broadcast.ShmRingBuffer object at 0x7f68fc0e5220>, local_subscribe_port=35137, remote_subscribe_port=None) [repeated 2x across cluster]
(RayWorkerWrapper pid=65163) INFO 12-07 20:30:35 model_runner.py:1072] Starting to load model Qwen/Qwen2.5-0.5B-Instruct... [repeated 6x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] Error executing method load_model. This might cause deadlock in distributed execution. [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] Traceback (most recent call last): [repeated 20x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connection.py", line 208, in _new_conn [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     sock = connection.create_connection( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/connection.py", line 73, in create_connection [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     raise err [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     sock.connect(sa) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] socket.timeout: timed out [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]  [repeated 30x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] The above exception was the direct cause of the following exception: [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 843, in urlopen [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     response = self._make_request( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 466, in _make_request [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     raise new_e [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     self._validate_conn(conn) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     conn.connect() [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/connection.py", line 693, in connect [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     self.sock = sock = self._new_conn() [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     raise ConnectTimeoutError( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x7f1b08093f40>, 'Connection to huggingface.co timed out. (connect timeout=10)') [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/adapters.py", line 688, in send [repeated 20x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     resp = conn.urlopen( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     retries = retries.increment( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/urllib3/util/retry.py", line 519, in increment [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type] [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models/Qwen/Qwen2.5-0.5B-Instruct (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f1b08093f40>, 'Connection to huggingface.co timed out. (connect timeout=10)')) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] During handling of the above exception, another exception occurred: [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/worker/worker_base.py", line 472, in execute_method [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     return executor(*args, **kwargs) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 334, in load_model [repeated 15x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     self.model_runner.load_model() [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     self.model = get_model(vllm_config=self.vllm_config) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/__init__.py", line 12, in get_model [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     return loader.load_model(vllm_config=vllm_config) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     model.load_weights(self._get_all_weights(model_config, model)) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 229, in load_weights [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     loader.load_weights(weights) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     autoloaded_weights = list(self._load_module("", self.module, weights)) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 181, in _load_module [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     for child_prefix, child_weights in self._groupby_prefix(weights): [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 103, in _groupby_prefix [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     for prefix, group in itertools.groupby(weights_by_parts, [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/models/utils.py", line 100, in <genexpr> [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     weights_by_parts = ((weight_name.split(".", 1), weight_data) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 313, in _get_all_weights [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     yield from self._get_weights_iterator(primary_weights) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 272, in _get_weights_iterator [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     hf_folder, hf_weights_files, use_safetensors = self._prepare_weights( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/loader.py", line 228, in _prepare_weights [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     hf_folder = download_weights_from_hf( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/vllm/model_executor/model_loader/weight_utils.py", line 234, in download_weights_from_hf [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     file_list = fs.ls(model_name_or_path, detail=False, revision=revision) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 280, in ls [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     resolved_path = self.resolve_path(path, revision=revision) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 186, in resolve_path [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     repo_and_revision_exist, err = self._repo_and_revision_exist(repo_type, repo_id, revision) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_file_system.py", line 121, in _repo_and_revision_exist [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     self._api.repo_info( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     return fn(*args, **kwargs) [repeated 10x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 2756, in repo_info [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     return method( [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/huggingface_hub/hf_api.py", line 2540, in model_info [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     r = get_session().get(path, headers=headers, timeout=timeout, params=params) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 602, in get [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     return self.request("GET", url, **kwargs) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]   File "/root/miniconda3/envs/pygong/lib/python3.9/site-packages/requests/sessions.py", line 589, in request [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     resp = self.send(prep, **send_kwargs) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     r = adapter.send(request, **kwargs) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     return super().send(request, *args, **kwargs) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480]     raise ConnectTimeout(e, request=request) [repeated 5x across cluster]
(RayWorkerWrapper pid=12143, ip=11.45.64.147) ERROR 12-07 20:30:45 worker_base.py:480] requests.exceptions.ConnectTimeout: (MaxRetryError("HTTPSConnectionPool(host='huggingface.co', port=443): Max retries exceeded with url: /api/models/Qwen/Qwen2.5-0.5B-Instruct (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f1b08093f40>, 'Connection to huggingface.co timed out. (connect timeout=10)'))"), '(Request ID: 6c7f6560-4b41-4311-a2f1-6f69e4c60184)') [repeated 5x across cluster]
[rank0]:[W1207 20:30:46.419931997 ProcessGroupNCCL.cpp:1250] Warning: WARNING: process group has NOT been destroyed before we destruct ProcessGroupNCCL. On normal program exit, the application should call destroy_process_group to ensure that any pending NCCL operations have finished in this process. In rare cases this process can exit before this point and block the progress of another member of the process group. This constraint has always been present,  but this warning has only been added since PyTorch 2.4 (function operator())
/root/miniconda3/envs/pygong/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 1 leaked shared_memory objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

</p>
</details> 

So why is it not connecting to huggingface.co? There were no such issues during the model download process. We hope for your assistance, thank you.

**Comment by jikunshang** [comment]
I feel this is a network issue. huggingface access is not stable in China. you can try with `vllm serve facebook/125m` to verify vllm/network functionality first.
or you can
1. download model manually, and use local model path. 
2. `export HF_ENDPOINT=https://hf-mirror.com`

**Comment by pygongnlp** [comment]
> I feel this is a network issue. huggingface access is not stable in China. you can try with `vllm serve facebook/125m` to verify vllm/network functionality first. or you can
> 
> 1. download model manually, and use local model path.
> 2. `export HF_ENDPOINT=https://hf-mirror.com`

I have already resolved the issue. The network proxy problem needs to be addressed before starting the Ray cluster, not after.

So thanks for your help ~

**Comment by zihaohe123** [comment]
Hi Peiyuan, 

I wonder if you have figured out how to perform multi-node offline inference? Thanks!
